flake.nix

```diff
@@ -28,27 +17,43 @@
     , haskellTar
     , flake-compat
     }:
-    let
-      name = "smoke";
-    in
     flake-utils.lib.eachDefaultSystem (system:
     let
       pkgs = import nixpkgs { inherit system; };
-      deps = import ./nix/deps.nix { inherit pkgs; };
-      extraLibs = import ./nix/libs.nix { inherit pkgs; };
-      ghc = import ./nix/ghc.nix {
-        inherit (pkgs) lib haskell;
-        ghcVersion = pkgs.lib.strings.fileContents ./ghc.version;
-        overrides = hself: hsuper: {
-          tar = hsuper.callCabal2nixWithOptions "tar" haskellTar "--no-check" { };
-        };
-      };
-      drv = import ./nix/smoke.nix { inherit ghc pkgs; };
+      inherit (pkgs) lib haskell;
+      inherit (haskell.lib) overrideCabal justStaticExecutables doStrip;
+
+      # required libraries with static linking enabled
+      staticLibs = with pkgs; [
```

Comment:
Why not pkgsStatic?

Reply:
Honestly, we can probably just remove this entirely. This comes from when I was hoping I could do the Linux build with Nix, but I gave up on that idea.

I was avoiding `pkgsStatic` because it just takes too long to build the whole world.